### ISSUE TITLE
Custom escape character option not working

### ISSUE THREAD

**Issue by divekarsc** [databricks/spark-csv#365]
I have been trying to save my data frame as a CSV.

df.write.format("com.databricks.spark.csv")
            .option("header", "true") // Use first line of all files as header
              .option("nullValue", "")
              .option("escape", null )
            .option("quoteMode", "MINIMAL")
            .save(outputName)

The option to set escape character to null or any other character is not working. It always picks \ (backslash) as escape character. 
I went through the code base and realized that spark-csv is in turn just calling withEscape(Character) method of CSVFormat in apache commons csv library, but somehow it isn't working.  

Note: My intention is, to write the DF as it is. I mean no character should have a special meaning as escape character and everything should be written as plain ascii text.

**Comment by HyukjinKwon** [comment]
Do you mind if I ask some sample input data, output and maybe desired output maybe?

**Comment by divekarsc** [comment]
Sure. Consider the following example.

Input:
{  "name": "BIOMED REALTY TRUST \r INC",  "sic_code": 6798,  "stock_symbol": "BMR" }
{  "name": "FATE THERAPEUTICS\n  INC",  "sic_code": 2836,  "stock_symbol": "FATE" }


Desired Output:
sic_code, stock_symbol, name
6798, BMR, BIOMED REALTY TRUST \r INC
2836, FATE, FATE THERAPEUTICS\n  INC

Actual output:

6798,BMR,BIOMED REALTY TRUST ^M INC
2836, FATE,FATE THERAPEUTICS
  INC

So basically, what I want is that the \ (backslash) character to lose its special meaning and be written as a simple ascii character in CSV too. So I tried setting it to some different character/null as read in Apache commons csv documentation. Dataframe reads it correctly, i.e. if I do df.show, it shows the text as it is.

**Comment by divekarsc** [comment]
@HyukjinKwon Don't know if you got a chance to look at this.

**Comment by divekarsc** [comment]
Hi,
I found something.
https://issues.apache.org/jira/browse/CSV-135?jql=project%20%3D%20CSV%20AND%20status%20in%20(Open%2C%20%22In%20Progress%22%2C%20Reopened)%20AND%20text%20~%20%22escape%22%20ORDER%20BY%20key%20DESC

May be this is a known issue with Apache Commons CSV 1.1. (In fact, 4-5 similar issues.) May be upgrade to a newer version of commons csv might help? :)

**Comment by petro-rudenko** [comment]
@divekarsc having the same issue. Updating csv-cmmons didn't help.

**Comment by sertaco** [comment]
@divekarsc do you know if this is resolved? 
Also I have issues with "disabling escape character" default '\' breaks my flow and passing none or null does not work. Anyone having any idea?
